hail/src/main/scala/is/hail/expr/ir/lowering/LowerDistributedSort.scala

```diff
@@ -176,17 +176,22 @@ object LowerDistributedSort {
                 ("perPartIntervalTuples", ApplyAggOp(Collect())(MakeTuple.ordered(Seq(GetField(dataRef, "min"), GetField(dataRef, "max")))))
               )), false)
           })) { aggResults =>
-            val sortedSamples = sortIR(flatMapIR(ToStream(GetField(aggResults, "samples"))) { onePartCollectedArray => ToStream(onePartCollectedArray)}) { case (left, right) =>
+            val sortedOversampling = sortIR(flatMapIR(ToStream(GetField(aggResults, "samples"))) { onePartCollectedArray => ToStream(onePartCollectedArray)}) { case (left, right) =>
               ApplyComparisonOp(LT(newKType), left, right)
             }
             val minArray = MakeArray(GetField(aggResults, "min"))
             val maxArray = MakeArray(GetField(aggResults, "max"))
             val tuplesInSortedOrder = tuplesAreSorted(GetField(aggResults, "perPartIntervalTuples"))
-            MakeStruct(Seq(
-              "pivotsWithEndpoints" -> ArrayFunctions.extend(ArrayFunctions.extend(minArray, sortedSamples), maxArray),
-              "isSorted" -> ApplySpecial("land", Seq.empty[Type], Seq(GetField(aggResults, "eachPartSorted"), tuplesInSortedOrder), TBoolean, ErrorIDs.NO_ERROR),
-              "intervalTuple" -> MakeTuple.ordered(Seq(GetField(aggResults, "min"), GetField(aggResults, "max")))
-            ))
+            bindIR(sortedOversampling) { sortedOversampling =>
+              val sortedSampling = ToArray(mapIR(StreamRange(0, ArrayLen(sortedOversampling), I32(oversamplingNum))) { idx =>
```

Comment:
I believe the range you want is `range(oversamplingNum - 1, length, oversamplingNum)`. For example, say `branchingFactor = 4` and `oversamplingNum = 3`. Then we collect `4*3-1 = 11` "oversamples", and we want to extract `4-1` evenly spaced samples:
```
[*, *, 2, *, *, 5, *, *, 8, *, *]
```